teuthology/kill.py

```diff
@@ -177,8 +177,8 @@ def kill_processes(run_name, pids=None):
             # Don't attempt to use sudo if it's not necessary
             proc_user = psutil.Process(int(pid)).username()
             if proc_user != getpass.getuser():
-                args.insert(0, 'sudo')
-            subprocess.call(args)
+                args = ['sudo', '-n'] + args
+            subprocess.check_call(args)
```

Comment:
Since we change to check_call, it means the kill command can be interrupted in the middle of sequential calls, like in killing several job, not even trying to kill the next job in a list. Do we really want this?

Reply:
It looks like it would fail if one of the processes already exited - could we check for ability to run sudo before this loop instead?

Reply:
Thanks for the feedback; I've re-implemented this, doing a check for `sudo`'s functionality before attempting to use it - and logging to inform the user of the choice we make if we can't use it.